### ISSUE TITLE
Terraform output error since 0.12.14 "Initialization required"

### ISSUE THREAD

**Issue by fozturner** [hashicorp/terraform#23370]
### Terraform Version

`Terraform v0.12.14`

### Terraform Configration Files

```hcl
output "output1" {
  value = "${module.modulename.valuename}"
}
```

### Expected behaviour

Output displayed successfully after a `terraform apply` is ran.

When running `terraform output` in the same directory where "apply" ran, the outputs should be outputted to screen.

### Actual behaviour

I get the following error when running my output command.

```
Initialization required. Please see the error message above.
```

### Steps to Reproduce

1. Config output in Terraform
2. `terraform init`
3. `terraform apply`
4. `terraform output`

### Additional context

This is running in an Azure DevOps pipeline. This worked in version 0.12.13, I haven't changed anything in the pipeline, only updated the TF version to 0.12.14.

When i try to run a `terraform init` again in the working directory it states that there are no configuration files, although if i run Get-ChildItem you can see the list of the files in the current directory.

**Comment by jbardin** [comment]
Hi @fozturner,

Sorry you've encountered an issue with the latest release. In order to troubleshoot this, it would be helpful to have a more complete configuration.

What is the backend you're using, and how is it configured?

It might help to show the `"backend"` block from your `.terraform/terraform.tfstate` file as well, but please make sure to sanitize any secrets that might be stored in there.

**Comment by fozturner** [comment]
Hey,

I have since deleted everything and reverted back to a 0.12.13 deployment of TF.

My backend file only consists of the below, so I assume that means it would not be present in my .tfstate? (I can't find anything referencing "backend")

```hcl
terraform {
  backend "azurerm" {
  }
}
```

I use this block of code to initialise in my Azure DevOps pipeline task:

```
/usr/local/bin/terraform init -input=false \
-backend-config="storage_account_name=$(terraform_backend_storage_account_name)" \
-backend-config="container_name=$(terraform_backend_container_name)" \
-backend-config="key=$(terraform_backend_key)" \
-backend-config="access_key=$STORAGE_KEY"
```

The rest of the configuration is below:

I have a deployment file that calls my keyvault module (sanitised):

```HCL
module "keyvault" {
  source                         = "../modules/keyvault/"
  key_vault_name                 = ""
  resourcegroup                  = ""
  costcode                       = ""
  environment                    = ""
  product                        = ""
  location                       = ""   
  subnetids                      = [
                                   ]
  sku                            = ""
  tenant_id                      = ""
  keyvault_object_id             = ""
}
```

At this deployment level i have this output:

```hcl
output "kvname" {
  value = "${module.keyvault.kvname}"
}
```

In my module:

```hcl
resource "azurerm_key_vault" "kvt" {
  name                            = "${var.key_vault_name}"
  resource_group_name             = "${var.resourcegroup}"
  location                        = "${var.location}"
  enabled_for_disk_encryption     = true
  enabled_for_template_deployment = true
  enabled_for_deployment          = true
  tenant_id                       = "${var.tenant_id}"

  access_policy {
    tenant_id    = "${var.tenant_id}"
    object_id    = "${var.keyvault_object_id}"

  key_permissions = [
    "get",
  ]

  secret_permissions = [
    "get",
    "list",
    "set",
  ]
  }
  sku_name = "${var.sku}"

  network_acls {
    default_action             = "Deny"
    bypass                     = "AzureServices"
    virtual_network_subnet_ids = "${var.subnetids}"
  }

  tags = {
    costcode    = "${var.costcode}"
    environment = "${var.environment}"
    product     = "${var.product}"
  }
}

output "kvname" {
  value = "${azurerm_key_vault.kvt.name}"
}
```

I then have another task that runs the `terraform output`.

**Comment by EddieOX** [comment]
I ran into this issue also. I fixed it by removing all the old 0.11 style syntax. For example, change:
` product     = "${var.product}"`
to:
` product     = var.product`

**Comment by Advin24** [comment]
I also ran into this exact issue. The fix mentioned by @EddieOX also fixed my problem.
